src/lib/crypto/elgamal.c

```diff
@@ -84,167 +84,117 @@
 
 #include "crypto/elgamal.h"
 #include "crypto/bn.h"
+#include "crypto/dsa.h"
+#include "crypto.h"
 
-int
-pgp_elgamal_public_encrypt_pkcs1(rng_t *                     rng,
-                                 uint8_t *                   g2k,
-                                 uint8_t *                   encm,
-                                 const uint8_t *             in,
-                                 size_t                      length,
-                                 const pgp_elgamal_pubkey_t *pubkey)
+#define ELGAMAL_MAX_P_BYTELEN BITS_TO_BYTES(DSA_MAX_P_BITLEN)
+
+rnp_result_t elgamal_encrypt_pkcs1(
+    rng_t* rng,
+    struct buf_t* g2k,
+    struct buf_t* encm,
+    const struct buf_t* in,
+    const pgp_elgamal_pubkey_t *pubkey)
 {
     botan_pubkey_t        key = NULL;
     botan_pk_op_encrypt_t op_ctx = NULL;
-    int                   ret = -1;
-    size_t                p_len = 0;
-    uint8_t *             bt_ciphertext = NULL;
+    rnp_result_t          ret = RNP_ERROR_BAD_PARAMETERS;
+    uint8_t               enc_buf[ELGAMAL_MAX_P_BYTELEN*2] = {0};   /* Max size of an output
+                                                                    len is twice an order of
+                                                                    underlying group (twice
+                                                                    byte-size of p) */
 
-    if (botan_mp_num_bytes(pubkey->p->mp, &p_len)) {
-        RNP_LOG("Wrong public key");
+    size_t tmp;
```

Comment:
It would be good to have a better name than `tmp` here. I get that it's reused later but I think it's more readable to just have separate variables with more descriptive names.

Reply:
it's used for different temporary things which are not worth having it's own variable defined